### ISSUE TITLE
No ES6 transforms

### ISSUE THREAD

**Issue by Psiablo** [reactjs/react-rails#359]
I have installed the gem as per the instructions.

I have a component file named `pinpad.js.jsx` in the components folder that makes use of several new ES6 features.

The resulting file sent to the browser is never transformed and therefore errors out on some browsers that don't support some of the used features.

I'm using things like arrow functions, object literals, and the `const` and `let` keywords.

I can't figure out why Babel is not doing anything.

I expect to write my file like so:
```
// Create PinPad
let Pinpad = React.createClass({
  propTypes: {
    url: React.PropTypes.string,
    onSuccess: React.PropTypes.func,
    onFailure: React.PropTypes.func,
    length: React.PropTypes.number,
    desc: React.PropTypes.string,
    name: React.PropTypes.string,
    showDesc: React.PropTypes.bool,
    showName: React.PropTypes.bool
  },

  getDefaultProps() {
    return {
      url: null,
      onSuccess: function(){ console.error('Pinpad: You have not supplied a onSuccess function.'); },
      onFailure: function(){ console.error('Pinpad: You have not supplied a onFailure function.'); },
      length: 4,
      name: 'No name',
      desc: 'Exit by entering your passcode.',
      showDesc: true,
      showName: true
    };
  },

  getInitialState() {
    return {
      passcode: '',
      hasError: false
    };
  },

  componentWillMount() {
    if (!this.props.url) {
      console.error('PINPAD: You have not supplied a url property. REQUIRED!');
    }
  },

  handlePasscodeUpdate(value) {
    // Handle "clear"
    if (value == 'clear') {
      this.resetPasscode();
      return;
    }

    // Handle "delete"
    if (value == 'delete') {
      this.setState({passcode: this.state.passcode.substring(0, this.state.passcode.length -1)});
      return;
    }

    // Else handle the #
    if (this.state.passcode.length < this.props.length) {
      var passcode = this.state.passcode + value;

      this.setState({passcode: passcode, hasError: false}, function() {
        // console.log("Passcode: " + this.state.passcode);

        if (this.state.passcode.length == this.props.length) {
          this.handleSubmit();
        }
      });
    } else {
      console.warn('Max passcode length already reached. Did we forget to reset after submit?');
    }
  },

  handleSubmit() {

    if(this.props.url) {
      $.ajax({
        type: 'POST',
        url: this.props.url,
        data: [{name: 'passcode', value: this.state.passcode}],
        success: (data) => {
          this.resetPasscode();
          this.props.onSuccess(data);
        },
        error: (jqxhr, status, error) => {
          this.resetPasscode();
          this.invalidCode();
          this.props.onFailure({message: jqxhr.responseText, status: jqxhr.status, statusText: error });
        }
      });
    } else {
      console.error('PINPAD: You have not supplied a url property. REQUIRED!');
      this.resetPasscode();
    }

  },

  resetPasscode() {
    this.setState({passcode: ''});
  },

  invalidCode() {
    this.setState({hasError: true});
  },

  render() {
    var classes = classNames({
      'Pinpad': true,
      'Pinpad--light': this.props.theme == 'light'
    });

    if (this.state.hasError) { classes += ' has-error'; }

    return (
      <div className={classes}>
        <div className="Pinpad-user">{this.props.name}</div>
        <div className="Pinpad-desc">{this.props.desc}</div>

        <PinpadDisplay length={this.props.length} enteredLength={this.state.passcode.length} />

        <div className="Pinpad-keys">
          <PinpadKey value="1" updatePasscode={this.handlePasscodeUpdate}/>
          <PinpadKey value="2" updatePasscode={this.handlePasscodeUpdate}/>
          <PinpadKey value="3" updatePasscode={this.handlePasscodeUpdate}/>
          <PinpadKey value="4" updatePasscode={this.handlePasscodeUpdate}/>
          <PinpadKey value="5" updatePasscode={this.handlePasscodeUpdate}/>
          <PinpadKey value="6" updatePasscode={this.handlePasscodeUpdate}/>
          <PinpadKey value="7" updatePasscode={this.handlePasscodeUpdate}/>
          <PinpadKey value="8" updatePasscode={this.handlePasscodeUpdate}/>
          <PinpadKey value="9" updatePasscode={this.handlePasscodeUpdate}/>
          <PinpadKey value="clear" classes="is-clear" updatePasscode={this.handlePasscodeUpdate}/>
          <PinpadKey value="0" updatePasscode={this.handlePasscodeUpdate}/>
          <PinpadKey value="delete" classes="is-delete" updatePasscode={this.handlePasscodeUpdate}/>
        </div>
      </div>
    );
  }
});


// Create PinPad Display
var PinpadDisplay = React.createClass({
  render() {
    var passcodeDisplay = [];
    for (var i = 0; i < this.props.length; i++) {
      var classes = '';

      // Determine if the particular key is "entered".
      if (i < this.props.enteredLength) {
        classes = 'is-entered';
      }

      passcodeDisplay.push(<li className={classes} key={i}></li>);
    }

    return (
      <ul className="Pinpad-display">
        {passcodeDisplay}
      </ul>
    );
  }
});


// Create Pinpad Key
var PinpadKey = React.createClass({
  handleClick(e) {
    this.props.updatePasscode(this.props.value);
  },

  render() {
    var classes = 'Pinpad-key';

    if (this.props.classes) {
      classes += ' ' + this.props.classes;
    }
    return (
      <button className={classes} onClick={this.handleClick}>{this.props.value}</button>
    );
  }
});
```

And I expect to get:
```
// Create PinPad
'use strict';

var Pinpad = React.createClass({
  displayName: 'Pinpad',

  propTypes: {
    url: React.PropTypes.string,
    onSuccess: React.PropTypes.func,
    onFailure: React.PropTypes.func,
    length: React.PropTypes.number,
    desc: React.PropTypes.string,
    name: React.PropTypes.string,
    showDesc: React.PropTypes.bool,
    showName: React.PropTypes.bool
  },

  getDefaultProps: function getDefaultProps() {
    return {
      url: null,
      onSuccess: function onSuccess() {
        console.error('Pinpad: You have not supplied a onSuccess function.');
      },
      onFailure: function onFailure() {
        console.error('Pinpad: You have not supplied a onFailure function.');
      },
      length: 4,
      name: 'No name',
      desc: 'Exit by entering your passcode.',
      showDesc: true,
      showName: true
    };
  },

  getInitialState: function getInitialState() {
    return {
      passcode: '',
      hasError: false
    };
  },

  componentWillMount: function componentWillMount() {
    if (!this.props.url) {
      console.error('PINPAD: You have not supplied a url property. REQUIRED!');
    }
  },

  handlePasscodeUpdate: function handlePasscodeUpdate(value) {
    // Handle "clear"
    if (value == 'clear') {
      this.resetPasscode();
      return;
    }

    // Handle "delete"
    if (value == 'delete') {
      this.setState({ passcode: this.state.passcode.substring(0, this.state.passcode.length - 1) });
      return;
    }

    // Else handle the #
    if (this.state.passcode.length < this.props.length) {
      var passcode = this.state.passcode + value;

      this.setState({ passcode: passcode, hasError: false }, function () {
        // console.log("Passcode: " + this.state.passcode);

        if (this.state.passcode.length == this.props.length) {
          this.handleSubmit();
        }
      });
    } else {
      console.warn('Max passcode length already reached. Did we forget to reset after submit?');
    }
  },

  handleSubmit: function handleSubmit() {
    var _this = this;

    if (this.props.url) {
      $.ajax({
        type: 'POST',
        url: this.props.url,
        data: [{ name: 'passcode', value: this.state.passcode }],
        success: function success(data) {
          _this.resetPasscode();
          _this.props.onSuccess(data);
        },
        error: function error(jqxhr, status, _error) {
          _this.resetPasscode();
          _this.invalidCode();
          _this.props.onFailure({ message: jqxhr.responseText, status: jqxhr.status, statusText: _error });
        }
      });
    } else {
      console.error('PINPAD: You have not supplied a url property. REQUIRED!');
      this.resetPasscode();
    }
  },

  resetPasscode: function resetPasscode() {
    this.setState({ passcode: '' });
  },

  invalidCode: function invalidCode() {
    this.setState({ hasError: true });
  },

  render: function render() {
    var classes = classNames({
      'Pinpad': true,
      'Pinpad--light': this.props.theme == 'light'
    });

    if (this.state.hasError) {
      classes += ' has-error';
    }

    return React.createElement(
      'div',
      { className: classes },
      React.createElement(
        'div',
        { className: 'Pinpad-user' },
        this.props.name
      ),
      React.createElement(
        'div',
        { className: 'Pinpad-desc' },
        this.props.desc
      ),
      React.createElement(PinpadDisplay, { length: this.props.length, enteredLength: this.state.passcode.length }),
      React.createElement(
        'div',
        { className: 'Pinpad-keys' },
        React.createElement(PinpadKey, { value: '1', updatePasscode: this.handlePasscodeUpdate }),
        React.createElement(PinpadKey, { value: '2', updatePasscode: this.handlePasscodeUpdate }),
        React.createElement(PinpadKey, { value: '3', updatePasscode: this.handlePasscodeUpdate }),
        React.createElement(PinpadKey, { value: '4', updatePasscode: this.handlePasscodeUpdate }),
        React.createElement(PinpadKey, { value: '5', updatePasscode: this.handlePasscodeUpdate }),
        React.createElement(PinpadKey, { value: '6', updatePasscode: this.handlePasscodeUpdate }),
        React.createElement(PinpadKey, { value: '7', updatePasscode: this.handlePasscodeUpdate }),
        React.createElement(PinpadKey, { value: '8', updatePasscode: this.handlePasscodeUpdate }),
        React.createElement(PinpadKey, { value: '9', updatePasscode: this.handlePasscodeUpdate }),
        React.createElement(PinpadKey, { value: 'clear', classes: 'is-clear', updatePasscode: this.handlePasscodeUpdate }),
        React.createElement(PinpadKey, { value: '0', updatePasscode: this.handlePasscodeUpdate }),
        React.createElement(PinpadKey, { value: 'delete', classes: 'is-delete', updatePasscode: this.handlePasscodeUpdate })
      )
    );
  }
});

// Create PinPad Display
var PinpadDisplay = React.createClass({
  displayName: 'PinpadDisplay',

  render: function render() {
    var passcodeDisplay = [];
    for (var i = 0; i < this.props.length; i++) {
      var classes = '';

      // Determine if the particular key is "entered".
      if (i < this.props.enteredLength) {
        classes = 'is-entered';
      }

      passcodeDisplay.push(React.createElement('li', { className: classes, key: i }));
    }

    return React.createElement(
      'ul',
      { className: 'Pinpad-display' },
      passcodeDisplay
    );
  }
});

// Create Pinpad Key
var PinpadKey = React.createClass({
  displayName: 'PinpadKey',

  handleClick: function handleClick(e) {
    this.props.updatePasscode(this.props.value);
  },

  render: function render() {
    var classes = 'Pinpad-key';

    if (this.props.classes) {
      classes += ' ' + this.props.classes;
    }
    return React.createElement(
      'button',
      { className: classes, onClick: this.handleClick },
      this.props.value
    );
  }
});
```

What am I missing or doing wrong? Why am I getting no babel transforms?

**Comment by rmosolgo** [comment]
Hmm, gotta try the easy one first: did you restart your local Rails server after installing the gem? The transform is hooked up during initialization, so you have to restart your app (eg, killing `rails server` and restarting it)

**Comment by Psiablo** [comment]
Yes. Rails server has been restarted numerous times.

**Comment by rmosolgo** [comment]
Hmm, I'm stumped then! It has a `.jsx` ending and the file looks right. It looks like that should work! Could you share some info:

- Gemfile.lock
- React configs from your app

**Comment by Psiablo** [comment]
```
# config/environments/development.rb
# React
  config.react.variant = :development
```

```
# config/environments/production.rb
# React version
  config.react.variant = :production
```

Currently nothing in `config/application.rb`, but I even tried adding
```
config.react.jsx_transform_options = {
  # various options supplied for testing, both blacklist and whitelist
}
```

Still nothing.

From `Gemfile.lock`:
```
react-rails (1.0.0)
      coffee-script-source (~> 1.8)
      connection_pool
      execjs
      rails (>= 3.1)
      react-source (~> 0.13)
    react-source (0.13.3)
```

Is there anything I can do in Rails console to check that babel is initialized correctly?

**Comment by Psiablo** [comment]
It should also be noted that the JSX transform is happening, but none of the es6 and default babel transforms I would expect take place.

For instance, all the JSX parts of the code are converted to their correct JS equivalents, like `React.createElement()`.

**Comment by rmosolgo** [comment]
Ah, I see. You're on `react-rails 1.0.0`, but  `BabelTransformer` was added in `1.1.0`: https://github.com/reactjs/react-rails/blob/master/CHANGELOG.md#new-features-4

So, it's using `JSXTransformer`, which explains why JSX is transformed but ES6 isn't!

You should update `react-rails` to at least 1.1.0 to use Babel transforms.

**Comment by Psiablo** [comment]
Just realized that to looking through the Changelog. Gracias.

**Comment by gnapse** [comment]
Hi, I'm using react-rails 1.3.1, and I just noticed my recent addition of react to the app does not work in Safari. I had been developing on Chrome all along, and since Chrome supports arrow functions, it has been working fine. In Safari I got an error, and it points me to incorrect syntax around an arrow function. So the source files are not being transpiled from ES6, although jsx syntax is being transpiled.

Any ideas?
